lib/biker.rb

```diff
@@ -1,3 +1,43 @@
 class Biker
+    attr_reader :name,
+                :max_distance,
+                :rides,
+                :acceptable_terrain
+
+    def initialize(name, max_distance)
+        @name = name
+        @max_distance = max_distance
+        @rides = {}
+        @acceptable_terrain = []
+    end
+
+    def learn_terrain!(terrain)
+        @acceptable_terrain << terrain
+    end
+
+    def log_ride(ride, ride_distance)
+       if @rides.key?(ride)
```

Comment:
This is sound logic! You're just missing the edge case conditions: if a biker doesn't know the terrain, or the ride distance is longer than their max_distance, they shouldn't log the ride. So you could do a nested if statement here, or try adding a guard clause at the beginning of the method like . . . 
`return nil unless . . . #code to check terrain and max distance`